internal/junit/junit.go

```diff
@@ -72,3 +75,42 @@ func Parse(data []byte) (TestSuites, error) {
 
 	return tss, err
 }
+
+// GetFailedTests get failed test list from the JUnit report.
+func GetFailedTests(report TestSuites) []string {
+	classes := map[string]bool{}
+	for _, s := range report.TestSuites {
+		for _, tc := range s.TestCases {
```

Comment:
Technically a code smell: Method receives an object that it has to unpack multiple times to get to the data that it actually wants. Because the method clearly does not care about the report or the suites and really just the test cases.

Reply:
I don't get it. The suites structure is just like TestSuites -> TestSuite -> TestCase -> className/testName. How can I get the className and testName without unpacking it?

Reply:
You have to unpack it, but it's of no concern to this method 😄 You have to unpack it before you call this method. See [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns).